src/client.js

```diff
@@ -103,11 +104,11 @@ export class Saturn {
       try {
         res = await fetch(parseUrl(url), { signal: controller.signal, ...options })
         clearTimeout(connectTimeout)
-        return { res, url, controller }
+        return { res, url, node, controller }
       } catch (err) {
-        throw new Error(
-          `Non OK response received: ${res.status} ${res.statusText}`
-        )
+        const error = err
+        error.node = node
+        throw new Error(error)
```

Comment:
I'm not sure what happens if you pass an object to the `Error` constructor, it expects a string.
```suggestion
        err.node = node
        throw err
```